src/FakeClient/GraphicsCaptureEventProcessor.h

```diff
@@ -0,0 +1,230 @@
+  // can overlap, so the time is not account for multiple times.
+  //
+  // This function also generates:
+  // - The corresponding GPU frame time histogram, that is divided in 1024 buckets, each bucket X
+  // represents how many frames have a duration between [X, X+1) ms.
+  // - Average GPU frame times[ms]
+  void CalculateGPUTimes() {
+    LOG("Calculating GPU Times");
+    uint64_t total_duration_ns = 0;
+    size_t current_submission = 0;
+    gpu_time_distribution_.fill(0u);
+    for (uint64_t next_frame_start_timestamp : frame_start_boundary_timestamps_) {
+      uint64_t frame_time_ns = 0;
+      std::vector<std::pair</*begin*/ uint64_t, /*end*/ uint64_t>> command_buffer_timestamps;
+      while (current_submission < submissions_.size()) {
+        const orbit_grpc_protos::GpuQueueSubmission& submission = submissions_[current_submission];
+        if (submission.meta_info().pre_submission_cpu_timestamp() >= next_frame_start_timestamp) {
+          break;
+        }
+
+        for (const orbit_grpc_protos::GpuSubmitInfo& submit_info : submission.submit_infos()) {
+          for (const orbit_grpc_protos::GpuCommandBuffer& command_buffer :
+               submit_info.command_buffers()) {
+            command_buffer_timestamps.emplace_back(std::make_pair(
+                command_buffer.begin_gpu_timestamp_ns(), command_buffer.end_gpu_timestamp_ns()));
+          }
+        }
+        ++current_submission;
+      }
+
+      sort(command_buffer_timestamps.begin(), command_buffer_timestamps.end());
+      uint64_t current_range_end = 0;
+      for (const std::pair<uint64_t, uint64_t>& command_buffer_timestamp :
+           command_buffer_timestamps) {
+        uint64_t begin_timestamp_ns = command_buffer_timestamp.first;
+        uint64_t end_timestamp_ns = command_buffer_timestamp.second;
+        if (begin_timestamp_ns >= current_range_end) {
+          frame_time_ns += (end_timestamp_ns - begin_timestamp_ns);
+          current_range_end = end_timestamp_ns;
+        } else if (end_timestamp_ns > current_range_end) {
+          frame_time_ns += end_timestamp_ns - current_range_end;
+          current_range_end = end_timestamp_ns;
+        }
+      }
+
+      if (!command_buffer_timestamps.empty()) {
+        double frame_time_ms = frame_time_ns / 1.0e6;
+        int distribution_index = std::min(static_cast<int>(frame_time_ms), kMaxTimeMs);
+        ++gpu_time_distribution_[distribution_index];
+      }
+
+      total_duration_ns += frame_time_ns;
+    }
+
+    gpu_avg_frame_time_ms = (total_duration_ns / frame_start_boundary_timestamps_.size()) / 1.0e6;
+  }
+
+  // Calculate the CPU frame times in [ms] as the time between sequential vkQueuePresentKHR
+  // calls. This function also generates:
+  // - The corresponding CPU frame time histogram, that is divided in 1024 buckets, where each
+  // bucket X represents how many frames have a duration between [X, X+1) ms.
```

Comment:
Why 1024? Since this is time, I expect something round like one second rather than a power of two.

Reply:
No particular reason, I maintained what the old tool had